pkg/staticnetworkconfig/generator.go

```diff
@@ -308,6 +309,67 @@ func (s *StaticNetworkConfigGenerator) validateInterfaceNamesExistence(macInterf
 	return nil
 }
 
+func (s *StaticNetworkConfigGenerator) validateInterfaceNamesExistenceYAML(macInterfaceMap models.MacInterfaceMap, networksYaml string) error {
+	interfaceNames := lo.Map(macInterfaceMap, func(m *models.MacInterfaceMapItems0, _ int) string { return m.LogicalNicName })
+
+	var config map[string]interface{}
+
+	// Unmarshal the JSON string into the config struct
+	err := yaml.Unmarshal([]byte(networksYaml), &config)
+	if err != nil {
+		s.log.WithError(err).Errorf("Error unmarshalling JSON")
+		return err
+	}
+
+	// See 'man systemd.net-naming-scheme' for interface naming protocol
+	predicatableIfaceNamePattern := regexp.MustCompile("^en[PsvxXbucaipod]")
+	Interfaces := config["interfaces"].([]interface{})
+
+	for _, iface := range Interfaces {
+		nic := iface.(map[interface{}]interface{})
+		interfaceName := nic["name"]
+		interfaceType := nic["type"]
+		switch interfaceType {
+		case "802-3-ethernet", "ethernet":
+			if !lo.Contains(interfaceNames, interfaceName.(string)) {
+				if !predicatableIfaceNamePattern.MatchString(interfaceName.(string)) {
```

Comment:
We also need to ignore the missing name when `nic["identifier"] == "mac-address"` (or perhaps `identifier, ok := nic["identifier"]; ok && identifier != "name"`)